tests/trainers/test_tasks.py

```diff
@@ -9,30 +9,37 @@
 from _pytest.monkeypatch import MonkeyPatch
 from omegaconf import OmegaConf
 
-from torchgeo.trainers import ClassificationTask, So2SatDataModule
+from torchgeo.trainers import (
+    BigEarthNetDataModule,
+    ClassificationTask,
+    MultiLabelClassificationTask,
+    So2SatDataModule,
+)
 
 from .test_utils import mocked_log
 
 
-@pytest.fixture(scope="module", params=[("rgb", 3), ("s2", 10)])
-def bands(request: SubRequest) -> Tuple[str, int]:
-    return cast(Tuple[str, int], request.param)
-
-
-@pytest.fixture(scope="module", params=[True, False])
-def datamodule(bands: Tuple[str, int], request: SubRequest) -> So2SatDataModule:
-    band_set = bands[0]
-    unsupervised_mode = request.param
-    root = os.path.join("tests", "data", "so2sat")
-    batch_size = 2
-    num_workers = 0
-    dm = So2SatDataModule(root, batch_size, num_workers, band_set, unsupervised_mode)
-    dm.prepare_data()
-    dm.setup()
-    return dm
-
-
 class TestClassificationTask:
+    @pytest.fixture(params=[("rgb", 3), ("s2", 10)])
+    def bands(self, request: SubRequest) -> Tuple[str, int]:
+        return cast(Tuple[str, int], request.param)
+
+    @pytest.fixture(params=[True, False])
+    def datamodule(
```

Comment:
I'm not sure if we actually need to parametrize the datamodule by bands or unsupervised mode. We already parametrize the datamodule during datamodule testing, here we're only trying to ensure that `ClassificationTask` works as expected. This will cut down on total testing time by a lot, especially since we're already parametrizing on loss and weight init.

Reply:
We need to test by bands because the min/max tensors we use for normalizing change based on the bands in the datamodule. If we create a base Classification DataModule we can test `unsupervised_mode` on that and not have to repeat for every datamodule.

Reply:
Just a heads up, I used `tests/trainers/test_so2sat.py` as a template for this so that will need to refactored later as well.